net/Socket.cpp

```diff
@@ -1553,7 +1553,7 @@ bool StreamSocket::checkRemoval(std::chrono::steady_clock::time_point now)
                                                     << ", " << *this);
         if (_socketHandler)
         {
-            _socketHandler->onDisconnect();
+            ensureDisconnected();
             if (!isShutdown())
```

Comment:
We could move this check+warning inside `ensureDisconnected()`, since we expect when we call `onDisconnect()` to have `isShutdown()` return true.